packages/dd-trace/src/profiling/profilers/wall.js

```diff
@@ -17,29 +17,61 @@ const TRACE_ENDPOINT_LABEL = 'trace endpoint'
 
 const { isWebServerSpan, endpointNameFromTags, getStartedSpans } = require('../webspan-utils')
 
-const beforeCh = dc.channel('dd-trace:storage:before')
+let beforeCh
 const enterCh = dc.channel('dd-trace:storage:enter')
 const spanFinishCh = dc.channel('dd-trace:span:finish')
 const profilerTelemetryMetrics = telemetryMetrics.manager.namespace('profilers')
 
 const ProfilingContext = Symbol('NativeWallProfiler.ProfilingContext')
 
 let kSampleCount
+let kCPEDContextCount
 
 function getActiveSpan () {
   const store = storage('legacy').getStore()
   return store && store.span
 }
 
+function toBigInt (spanId) {
+  return spanId !== null && typeof spanId === 'object' ? spanId.toBigInt() : spanId
```

Comment:
I believe that would suffice in this case (either it is a ID or it is not defined)

```suggestion
  return spanId?.toBigInt()
```